src/renderer.ts

```diff
@@ -169,8 +169,10 @@ class Renderer extends EventEmitter<RendererEvents> {
     const div = document.createElement('div')
     const shadow = div.attachShadow({ mode: 'open' })
 
+    const nonce = this.options.nonce;
+
     shadow.innerHTML = `
-      <style>
+      <style ${nonce ? `nonce="${nonce}"` : ''}>
```

Comment:
This should be sanitized for XSS. Something like nonce.replace(/"/g, '')

Reply:
Better above where you do const nonce

Reply:
@katspaugh I've also added some type validation just to avoid any errors